mne/epochs.py

```diff
@@ -2628,9 +2628,17 @@ def add_annotations_to_metadata(self):
         # onsets, durations, and descriptions
         epoch_annot_list = self.get_annotations_per_epoch()
         onset, duration, description = [], [], []
-        for epoch in epoch_annot_list:
+        for epoch_annot in epoch_annot_list:
             for ix, annot_prop in enumerate((onset, duration, description)):
-                entry = [annot[ix] for annot in epoch] if len(epoch) else []
+                entry = [annot[ix] for annot in epoch_annot] \
+                    if len(epoch_annot) else []
```

Comment:
Sorry I missed this the first time -- I don't think we need the if here. If the length is zero, you should end up with an empty list anyway